### PR TITLE
Fix build on Mac OS

### LINKED ISSUE #15804: int_hashes_perf.cpp build is broken
This is caused by https://github.com/ClickHouse/ClickHouse/pull/15008 which remove the following codes, see [diffs](https://github.com/ClickHouse/ClickHouse/pull/15008/files#diff-7c684e619b06ba1d15887c5342cf6609)

```c++
static inline size_t _intHash64(UInt64 x)
{
    return static_cast<size_t>(intHash64(x));
}
```

the codes were introduced  by  https://github.com/ClickHouse/ClickHouse/pull/1356:

> Basically, in the current version of Poco, unsigned long no longer aliases to UInt64 with LP64 pocoproject/poco#1870 (or any other type, unsigned long is basically not convertible to anything).
> 
> The size_t aliases to unsigned long with clang, so all the uses of size_t instead of UInt64 when interacting with Poco interfaces are gone pocoproject/poco#1147. I replaced uses with UInt64 where it makes sense, and added an overloaded function for readVarUInt() to support size_t.

-----
Make sure that `git diff` result is empty and you've just pulled fresh master. Try cleaning up cmake cache. Just in case, official build instructions are published here: https://clickhouse.yandex/docs/en/development/build/

**Operating system**
macos

**Cmake version**
cmake version 3.16.2

**Ninja version**

**Compiler name and version**
Apple clang version 11.0.3 (clang-1103.0.32.29)

**Full cmake and/or ninja output**
```
[100%] Building CXX object src/Common/tests/CMakeFiles/int_hashes_perf.dir/int_hashes_perf.cpp.o
/ClickHouse/src/Common/tests/int_hashes_perf.cpp:282:33: error: no matching function for call to 'test'
    if (!method || method == 3) test<intHash64> (n, data.data(), "2: intHash64");
                                ^~~~~~~~~~~~~~~
/ClickHouse/src/Common/tests/int_hashes_perf.cpp:209:20: note: candidate template ignored: invalid explicitly-specified argument for template parameter 'Func'
static inline void test(size_t n, const UInt64 * data, const char * name)
```

### PR DESCRIPTION
**Pull request by alexey-milovidov**
Changelog category (leave one):
- Build/Testing/Packaging Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix build of one miscellaneous example tool on Mac OS. Note that we don't build examples on Mac OS in our CI (we build only ClickHouse binary), so there is zero chance it will not break again. This fixes #15804.

### CODE REVIEW

File: src/Common/tests/int_hashes_perf.cpp [hunk 1]
```diff
@@ -279,7 +279,9 @@ int main(int argc, char ** argv)
 
     if (!method || method == 1) test<identity>  (n, data.data(), "0: identity");
     if (!method || method == 2) test<intHash32> (n, data.data(), "1: intHash32");
+#if !defined(__APPLE__) /// The difference in size_t: unsigned long on Linux, unsigned long long on Mac OS.
```
Comment (baibaichen) [inline src/Common/tests/int_hashes_perf.cpp:282]: why can't revert the removing _intHash64?

Comment (alexey-milovidov) [reply src/Common/tests/int_hashes_perf.cpp:282]: It's also possible, but the presence of this function is surprising to the reader.

Comment (alexey-milovidov) [reply src/Common/tests/int_hashes_perf.cpp:282]: Now you cannot run one special case of one miscellaneous benchmarking tool on Mac OS... I think it's passable.

Comment (alexey-milovidov) [reply src/Common/tests/int_hashes_perf.cpp:282]: This benchmarking tool initially served for fine tuning of the algorithms... but now it remains just for the purpose of demonstration and experiments.

Comment (alexey-milovidov) [reply src/Common/tests/int_hashes_perf.cpp:282]: BTW, intHash64 is exactly the same as `murmurMix`.

Comment (baibaichen) [reply src/Common/tests/int_hashes_perf.cpp:282]: > BTW, intHash64 is exactly the same as `murmurMix`.

If so,  I think it's better to remove this benchmarking for all platform for consistency.

**Pull request merged**